src/apps/MASSVEC3DPA-Cuda.cpp

```diff
@@ -0,0 +1,855 @@
+                              ); // RAJA::loop<inner_x>
+                            } // lambda (qy)
+                          ); // RAJA::loop<inner_y>
+                        } // lambda (qz)
+                      ); // RAJA::loop<inner_z>
+
+                      ctx.teamSync();
+
+                      RAJA::loop<inner_z>(ctx, RAJA::RangeSegment(0, MVPA_Q1D),
+                        [&](Index_type qz) {
+                          RAJA::loop<inner_y>(ctx, RAJA::RangeSegment(0, MVPA_D1D),
+                            [&](Index_type dy) {
+                              RAJA::loop<inner_x>(ctx, RAJA::RangeSegment(0, MVPA_D1D),
+                                [&](Index_type dx) {
+                                  MASSVEC3DPA_7;
+                                } // lambda (dx)
+                              ); // RAJA::loop<inner_x>
+                            } // lambda (dy)
+                          ); // RAJA::loop<inner_y>
+                        } // lambda (qz)
+                      ); // RAJA::loop<inner_z>
+
+                      ctx.teamSync();
+
+                      RAJA::loop<inner_z>(ctx, RAJA::RangeSegment(0, MVPA_D1D),
+                        [&](Index_type dz) {
+                          RAJA::loop<inner_y>(ctx, RAJA::RangeSegment(0, MVPA_D1D),
+                            [&](Index_type dy) {
+                              RAJA::loop<inner_x>(ctx, RAJA::RangeSegment(0, MVPA_D1D),
+                                [&](Index_type dx) {
+                                MASSVEC3DPA_8;
+                                } // lambda (dx)
+                              ); // RAJA::loop<inner_x>
+                            } // lambda (dy)
+                          ); // RAJA::loop<inner_y>
+                        } // lambda (dz)
+                      ); // RAJA::loop<inner_z>
+
+                      ctx.teamSync();
+
+                    } // c - dim loop
+                  }   // lambda (e)
+              );      // RAJA::loop<outer_x>
+            }         // outer lambda (ctx)
+        );            // RAJA::launch
+
+      } // loop over kernel reps
+      stopTimer();
+    }
+
+    break;
+  }
+
+  default: {
+
+    getCout() << "\n MASSVEC3DPA : Unknown Cuda variant id = " << vid
+              << std::endl;
+    break;
+  }
+  }
+}
+
+void MASSVEC3DPA::runCudaVariant(VariantID vid, size_t tune_idx) {
+
+  seq_for(gpu_block_sizes_type{}, [&](auto block_size) {
+    setBlockSize(block_size);
+    runCudaVariantImpl<block_size>(vid, tune_idx);
+  });
+}
+
+void MASSVEC3DPA::setCudaTuningDefinitions(VariantID vid) {
+
+  seq_for(gpu_block_sizes_type{}, [&](auto block_size) {
+    if (run_params.numValidGPUBlockSize() == 0u ||
+        run_params.validGPUBlockSize(block_size)) {
+
+      if (vid == RAJA_CUDA) {
+        addVariantTuningName(vid, "BLOCKDIM_LOOP_INC");
+        addVariantTuningName(vid, "COMPILE_LOOP_INC");
+        addVariantTuningName(vid, "DIRECT");
+      }
+
+      if (vid == Base_CUDA) {
```

Comment:
Base should be above RAJA like it is in the file.

Reply:
fixed.